core/chain_manager.go

```diff
@@ -800,8 +797,18 @@ func (self *ChainManager) merge(oldBlock, newBlock *types.Block) error {
 		PutTransactions(self.chainDb, block, block.Transactions())
 		PutReceipts(self.chainDb, GetBlockReceipts(self.chainDb, block.Hash()))
 
+		addedTxs = append(addedTxs, block.Transactions()...)
+	}
+
+	var diff types.Transactions
+	types.TxDifference(deletedTxs, addedTxs)
```

Comment:
```types.TxDifference``` returns a new slice as the set different of a to b, but does not modify a or b. Looks like it's return should be assigned here.